apps/shipments/serializers.py

```diff
@@ -234,14 +234,49 @@ def validate_delivery_act(self, delivery_act):
 
 class PermissionLinkSerializer(serializers.ModelSerializer):
     shipment_id = serializers.CharField(max_length=36, required=False)
+    emails = serializers.ListField(
+        child=serializers.EmailField(),
+        min_length=0,
+        required=False
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(PermissionLinkSerializer, self).__init__(*args, **kwargs)
+        self.subject = None
+        self.link = None
 
     class Meta:
         model = PermissionLink
-        exclude = ('shipment', )
+        exclude = ('shipment',)
 
-    def validate(self, attrs):
-        attrs['shipment_id'] = self.context['shipment_id']
-        return attrs
+    def create(self, validated_data):
+        username = getattr(self.context['user'], 'username', None)
```

Comment:
Maybe wrap this and the subject/link part in a if settings.PROFILES_ENABLED as it shouldn't really be hit otherwise.